### PR TITLE
Planning: add backside caution-level vehicle to St-boundary

### PR DESCRIPTION
**Pull request by JasonXu12**

### CODE REVIEW

File: modules/planning/common/obstacle.h [hunk 1]
```diff
@@ -124,6 +124,10 @@ class Obstacle {
 
   static bool IsValidTrajectoryPoint(const common::TrajectoryPoint& point);
 
+  const bool IsCautionLevelObstacle() const {
+      return is_caution_level_obstacle_;
+    }
+
```
Comment (xiaoxq) [inline modules/planning/common/obstacle.h]: Fix indents.

File: modules/planning/common/reference_line_info.cc [hunk 2]
```diff
@@ -385,6 +385,8 @@ bool ReferenceLineInfo::AddObstacles(
 }
 
 bool ReferenceLineInfo::IsUnrelaventObstacle(const Obstacle* obstacle) {
+  if (obstacle->IsCautionLevelObstacle())
+    return false;
```
Comment (xiaoxq) [inline modules/planning/common/reference_line_info.cc]: Always apply brackets to scopes.

File: modules/planning/traffic_rules/backside_vehicle.cc [hunk 3]
```diff
@@ -36,8 +36,9 @@ void BacksideVehicle::MakeLaneKeepingObstacleDecision(
       adc_sl_boundary.end_s() - adc_sl_boundary.start_s();
   for (const auto* obstacle : path_decision->obstacles().Items()) {
     if (obstacle->PerceptionSLBoundary().end_s() >=
-        adc_sl_boundary.end_s()) {  // don't ignore such vehicles.
-      continue;
+        adc_sl_boundary.end_s() || obstacle->IsCautionLevelObstacle()) {
+        // don't ignore such vehicles.
+        continue;
```
Comment (xiaoxq) [inline modules/planning/traffic_rules/backside_vehicle.cc]: Indents as 2.

Comment (JasonXu12) [reply modules/planning/traffic_rules/backside_vehicle.cc]: Fixed. Thanks!

File: modules/planning/common/reference_line_info.cc (hunk 2 above)
Comment (JasonXu12) [reply modules/planning/common/reference_line_info.cc]: Fixed. Thanks!

File: modules/planning/common/obstacle.h (hunk 1 above)
Comment (JasonXu12) [reply modules/planning/common/obstacle.h]: Fixed. Thanks!

File: modules/planning/common/obstacle.h [hunk 4]
```diff
@@ -124,6 +124,10 @@ class Obstacle {
 
   static bool IsValidTrajectoryPoint(const common::TrajectoryPoint& point);
 
+  const bool IsCautionLevelObstacle() const {
```
Comment (xiaoxq) [inline modules/planning/common/obstacle.h]: 1. Don't return a `const` copy which has no benefit but harm.
2. Better to declare `inline` for functions in header file, or you may run into "redefinition" error.

Comment (JasonXu12) [reply modules/planning/common/obstacle.h]: Fixed. Thanks!

**Pull request merged**